pulp_container/app/registry_api.py

```diff
@@ -645,10 +646,14 @@ def put(self, request, path, pk=None):
             upload = models.Upload.objects.get(pk=pk, repository=repository)
         except models.Upload.DoesNotExist as e_upload:
             # Upload has been deleted => task has started or even finished
-            task = Task.objects.filter(
-                name__endswith="add_and_remove",
-                reserved_resources_record__resource=f"upload:{pk}",
-            ).last()
+            task = (
+                Task.objects.filter(name__endswith="add_and_remove")
+                .filter(
+                    Q(reserved_resources_record__contains=[f"shared:upload:{pk}"])
+                    | Q(reserved_resources_record__contains=[f"upload:{pk}"])
```

Comment:
Is this really needed? Looking at
https://github.com/pulp/pulp_container/pull/868/files#diff-1b944e15118f82233aa03a435e9e05c188ad0d371b71be43260506b4a5eeddffL705
We never create the "shared:" version with this version of pulp_container.

Reply:
2.8 is compat with 3.14 and 3.15

Reply:
```
[ipanova@puffy pulpcore]$ git grep 'shared:'
pulpcore/tasking/pulpcore_worker.py:                    if not resource.startswith("shared:")
pulpcore/tasking/pulpcore_worker.py:                    if resource.startswith("shared:") and resource[7:] not in exclusive_resources
pulpcore/tasking/tasks.py:                (f"shared:{resource}" for resource in _validate_and_get_resources(shared_resources))
[ipanova@puffy pulpcore]$ git checkout 3.14
Switched to branch '3.14'
Your branch is up to date with 'upstream/3.14'.
[ipanova@puffy pulpcore]$ git grep 'shared:'
[ipanova@puffy pulpcore]$ 
```

Reply:
What i mean is that we declare the fake resource to be shared in the same file (at the dispatch some lines further down) [not on this branch]. And as long as we cannot _use_ shared resources on this branch for compatibility, we don't need to expect the fake resource to be listed shared, even if the version of pulpcore in use allows for it.

Reply:
i see, yes you are right